kendra-aaron/scripts/articleView.js

```diff
@@ -0,0 +1,121 @@
+
+articleView.populateFilters = () => {
+  $('article').each(function() {
+    if (!$(this).hasClass('template')) {
+      let val = $(this).find('address a').text();
+      let optionTag = `<option value="${val}">${val}</option>`;
+
+      if ($(`#author-filter option[value="${val}"]`).length === 0) {
+        $('#author-filter').append(optionTag);
+      }
+
+      val = $(this).attr('data-category');
+      optionTag = `<option value="${val}">${val}</option>`;
+      if ($(`#category-filter option[value="${val}"]`).length === 0) {
+        $('#category-filter').append(optionTag);
+      }
+    }
+  });
+};
+
+articleView.handleAuthorFilter = () => {
+  $('#author-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-author="${$(this).val()}"]`).fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = () => {
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-category="${$(this).val()}"]`).fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#author-filter').val('');
+  });
+};
+
+articleView.handleMainNav = () => {
+  $('nav').on('click', '.tab', function(e) {
+    e.preventDefault();
+    $('.tab-content').hide();
+    $(`#${$(this).data('content')}`).fadeIn();
+  });
+
+  $('nav .tab:first').click();
+};
+
+articleView.setTeasers = () => {
+  $('.article-body *:nth-of-type(n+2)').hide();
+  $('article').on('click', 'a.read-on', function(e) {
+    e.preventDefault();
+    if ($(this).text() === 'Read on →') {
+      $(this).parent().find('*').fadeIn();
+      $(this).html('Show Less &larr;');
+    } else {
+      $('body').animate({
+        scrollTop: ($(this).parent().offset().top)
+      },200);
+      $(this).html('Read on &rarr;');
+      $(this).parent().find('.article-body *:nth-of-type(n+2)').hide();
+    }
+  });
+};
+
+articleView.initNewArticlePage = () => {
+  $('.tab-content').show();
+  $('#export-field').hide();
+  $('#article-json').on('focus', function(){
+    this.select();
+  });
+
+  $('#new-form').on('change', 'input, textarea', articleView.create);
+};
+
+articleView.create = () => {
+  let article;
+  $('#articles').empty();
+
+  article = new Article({
+    title: $('#article-title').val(),
+    author: $('#article-author').val(),
+    authorUrl: $('#article-author-url').val(),
+    category: $('#article-category').val(),
+    body: $('#article-body').val(),
+    publishedOn: $('#article-published:checked').length ? new Date() : null
+  });
+
+  $('#articles').append(article.toHtml());
+  $('pre code').each(function(i, block) {
+    hljs.highlightBlock(block);
+  });
+
+  $('#export-field').show();
+  $('#article-json').val(`${JSON.stringify(article)},`);
+};
+
+
+articleView.initIndexPage = () => {
```

Comment:
articleView.js should not have been changed at all. `Article.fetchAll();` gets called first and runs `articleView.initIndexPage` as part of its functionality